README.md

```diff
@@ -1,16 +1,61 @@
 # Overview
-This package can be added to your workspaces to provide job runners
-that can fulfill resources requested by plans.
+This package can be used to create a wake `runner` that will set environment
+variables for jobs from JSON values based on the resources they request. A
+default runner is also included with this package.
 
 # Resources
-Resources describe a tool that is requested by a job. They are strings
-that have the format `{vendor}/{tool}/{version}`. If there is no vendor
-then the format is `{tool}/{version}`.
+Resources describe a tool that is requested by a job. They are strings.
+This package expects resources to have the format `{vendor}/{tool}/{version}`.
+If there is no vendor then the format is `{tool}/{version}`.
 
-# Runners
-This package provides a runner that sets the `PATH` of wake
-jobs to the value of the `WAKE_PATH` environment variable if it is set
-by the user.
+# Usage
+To use this you need to create a JSON file that maps resources to a JSON object
+representing the environment variables you want to set when that resource is
+requested.
+```json
+{
+  "google": {
+    "protobuf": {
+      "3.5.1": {
+        "PATH": "/protobuf/bin"
+      }
+    }
+  },
+  "verilator": {
+    "4.008": {
+      "PATH": "/verilator/bin",
+      "VERILATOR_ROOT": "/verilator"
+    }
+  }
+}
+```
 
-This package also provides a runner that will provide the riscv-tools
-resource if the user has set the `RISCV` environment variable.
+Then you need to `publish` a list of `Path`s to your JSON files to `environmentJSONs`.
+```wake
+publish environmentJSONs = source "path/to/environment.json", Nil
+```
+
+Now when a wake job requests the resource `google/protobuf/>=3.5.0`,
+`/protobuf/bin` will be added to the `PATH` environment variable for that job.
+
+
+# Making your own runner
+This package defines the global function `makeJSONEnvironmentRunner` that takes
+a `JSONEnvironmentRunnerPlan` and returns a `Runner`. You can use this to make
+your own runner if you would like to use a different version matching function
+or a different set of JSONs. This also defines `semverMatches`, which is the
+semver matching function using this
+[python package](https://pypi.org/project/semver/).
```

Comment:
I don't think we can make the assumption that all other packages follow SemVer, so this may not be a good idea. In particular, SemVer mandates exactly three groups of digits (the groups that are delimited by periods). Not all packages follow a 3-group scheme, as some may have a 2-group or even a 4-group scheme.

Let me list a few examples:

[Ghostscript](https://www.ghostscript.com/release_history.html) uses just a 2-group versioning scheme, which Linux distros such as [Ubuntu](https://launchpad.net/ubuntu/+source/ghostscript) appear to respect.
[Cabal](https://www.haskell.org/cabal/download.html) (the Haskell package manager) uses a 4-group versioning scheme.

Reply:
This doesn't make that assumption though? The matching function is a parameter. the semver matcher is just the one that's used by the included runner

Reply:
Oh, sorry, upon first read it wasn't to clear to me that you could pass in a non-semver matcher, since it's very subtly mention. It's also kind of unintuitive that the actual `JSONEnvironmentRunnerPlan` tuple property is named `SemverMatchFn`.

Maybe I'm just bad at reading comprehension, but I think we could improve this documentation. If I were to try to go back and describe my thought process while reading this for the first time, the sentence

> You can use this to make your own runner if you would like to use a different version matching function

doesn't make a whole lot of sense on first read because that is the first time you introduce the concept of a "version matching function" in this document. My brain didn't have an anchor for what a "version matching function" was until the following sentence, so saying that you can "use a different version matching function" is difficult to comprehend because I don't even know what one is let alone why I would want to override it.